ui/app/src/components/PublishOnDemandWidget/PublishOnDemandWidget.tsx

```diff
@@ -118,30 +120,48 @@ const initialPublishGitFormData = {
 
 interface PublishOnDemandWidgetProps {
   siteId: string;
+  onSubmittingAndOrPendingChange?(value: onSubmittingAndOrPendingChangeProps): void;
 }
 
 export function PublishOnDemandWidget(props: PublishOnDemandWidgetProps) {
-  const { siteId } = props;
+  const { siteId, onSubmittingAndOrPendingChange } = props;
   const { classes } = useStyles();
   const dispatch = useDispatch();
   const { formatMessage } = useIntl();
   const [mode, setMode] = useState<PublishOnDemandMode>(null);
   const [isSubmitting, setIsSubmitting] = useState(false);
+  const [initialPublishingTarget, setInitialPublishingTarget] = useState(null);
   const [publishingTargets, setPublishingTargets] = useState(null);
   const [publishingTargetsError, setPublishingTargetsError] = useState(null);
   const [publishGitFormData, setPublishGitFormData] = useSpreadState<PublishFormData>(initialPublishGitFormData);
   const [publishGitFormValid, setPublishGitFormValid] = useState(false);
+  const publishGitFormChanged = useMemo(() => {
+    return (
+      publishGitFormData.commitIds !== initialPublishGitFormData.commitIds ||
+      publishGitFormData.comment !== initialPublishGitFormData.comment ||
+      publishGitFormData.environment !== initialPublishingTarget
+    );
+  }, [initialPublishingTarget, publishGitFormData]);
   const [publishStudioFormData, setPublishStudioFormData] =
     useSpreadState<PublishFormData>(initialPublishStudioFormData);
   const [publishStudioFormValid, setPublishStudioFormValid] = useState(false);
+  const publishStudioFormChanged = useMemo(() => {
+    return (
+      publishStudioFormData.path !== initialPublishStudioFormData.path ||
+      publishStudioFormData.comment !== initialPublishStudioFormData.comment ||
+      publishStudioFormData.environment !== initialPublishingTarget
+    );
+  }, [initialPublishingTarget, publishStudioFormData]);
```

Comment:
These change on every keystroke so not worth the memo

Reply:
you should check the mode and keep a single "hasChanges" that then use on the effect. Only run the calculations (without memo) for the applicable form.